src/parser.cc

```diff
@@ -1113,12 +1115,36 @@ void addNestedModel(ElementPtr _sdf, ElementPtr _includeSDF)
   for (std::map<std::string, std::string>::iterator iter = replace.begin();
        iter != replace.end(); ++iter)
   {
-    replace_all(str, std::string("\"") + iter->first + "\"",
-                std::string("\"") + iter->second + "\"");
-    replace_all(str, std::string("'") + iter->first + "'",
-                std::string("'") + iter->second + "'");
-    replace_all(str, std::string(">") + iter->first + "<",
-                std::string(">") + iter->second + "<");
+    std::string oldName(iter->first);
+    std::string nameWithNestedPrefix(iter->second);
+    replace_all(str, std::string("\"") + oldName + "\"",
+                     std::string("\"") + nameWithNestedPrefix + "\"");
+    replace_all(str, std::string("'") + oldName + "'",
+                     std::string("'") + nameWithNestedPrefix + "'");
+    replace_all(str, std::string(">") + oldName + "<",
+                     std::string(">") + nameWithNestedPrefix + "<");
+    // Deal with nested model inside other nested model and already with
+    // ::namespace:: entries in the name
+    replace_all(str, std::string(">") + oldName + "::",
+                     std::string(">") + nameWithNestedPrefix + "::");
+    // remove duplicate prefixes introduced by the logic above
+    // if the full name (iter.second) was used in the original SDF: (i.e
+    // iter.second::component)
+    std::string::size_type found = nameWithNestedPrefix.find(oldName);
+    if (found != std::string::npos)
+    {
+      std::string nestedPrefix = nameWithNestedPrefix;
+      nestedPrefix.erase(found, oldName.length());
+      replace_all(str, nestedPrefix + nestedPrefix, nestedPrefix);
```

Comment:
Is this still needed after removing some of the `replace_all` cases?

Reply:
Umm I think you are right, this is not needed anymore after reducing the cases to just one. Removed 0f7ccf8